Lib/test/test_gc.py

```diff
@@ -734,6 +734,10 @@ def test_get_stats(self):
         self.assertEqual(new[1]["collections"], old[1]["collections"])
         self.assertEqual(new[2]["collections"], old[2]["collections"] + 1)
 
+    def test_freeze(self):
+        gc.freeze()
```

Comment:
You should really undo the freezing after this test. We don't want tests to have so large side effects.

Reply:
Good point! This will require adding the "thaw" or "unfreeze" function that will move objects back from the permanent generation to a properly collected one. Do you think it would be enough for that call to move all of them to gen0? This won't be exactly "undoing the freeze" but doesn't require keeping track of which object belonged to which generation before.

Reply:
Anything that empties the permanent generation is good enough IMHO. People can call `gc.collect` explicitly afterwards if they want to collect those objects.

PS: as a non-native English locutor, I much prefer "unfreeze" to "thaw" :-)

Reply:
Oh, didn't notice your "general comment" until now. So it seems we agree that moving everything to a single generation makes sense. But you're suggesting the oldest generation (gen2) instead. Makes sense.

Reply:
ok, I was just too lazy to add it but fine will update..